src/main/java/org/elasticsearch/rest/action/search/RestClearScrollAction.java

```
@@ -63,4 +82,30 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         }
         return Strings.splitStringByCommaToArray(scrollIds);
     }
+
+    public static ClearScrollRequest buildFromContent(BytesReference content, ClearScrollRequest clearScrollRequest) throws ElasticsearchIllegalArgumentException {
+        try (XContentParser parser = XContentHelper.createParser(content)) {
+            XContentParser.Token token;
+            String currentFieldName = null;
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    currentFieldName = parser.currentName();
+                } else if (token == XContentParser.Token.START_ARRAY) {
+                    if ("scroll_id".equals(currentFieldName)) {
+                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                            if (token.isValue() == false) {
+                                throw new ElasticsearchIllegalArgumentException("scroll_id array element should only contain scroll_id");
+                            }
+                            clearScrollRequest.addScrollId(parser.text());
+                        }
+                    }
+                }
```

can we throw an exception if we ran into a type we don't know?